tests/conftest.py

```diff
@@ -165,6 +166,48 @@ def redis_result_backend():
     return backend
 
 
+@pytest.fixture
+def redis_state_backend():
+    backend = st_backends.RedisBackend()
+    check_redis(backend.client)
+    return backend
+
+
+@pytest.fixture
+def stub_state_backend():
+    return st_backends.StubBackend()
+
+
+@pytest.fixture
+def state_backends(redis_state_backend, stub_state_backend):
+    return {"redis": redis_state_backend, "stub": stub_state_backend}
+
+
+@pytest.fixture(params=["redis", "stub"])
+def state_backend(request, state_backends):
+    return state_backends[request.param]
+
+
+@pytest.fixture
+def state_middleware(state_backend):
+    broker = remoulade.get_broker()
+    middleware = MessageState(backend=state_backend)
+    broker.add_middleware(middleware)
+    return middleware
+
+
+@pytest.fixture
+def do_work():
+    broker = remoulade.get_broker()
+
+    @remoulade.actor
+    def do_work(x):
```

Comment:
There should not be an argument for do_work. It should be simplest possible. Use another actor for your last test.